src/components/MessageInput/__tests__/MessageInput.test.js

```diff
@@ -0,0 +1,479 @@
+      const formElement = await findByPlaceholderText(inputPlaceholder);
+      dropFile(
+        new File(['content'], filename, { type: 'text/plain' }),
+        formElement,
+      );
+
+      const filenameText = await findByText(filename);
+
+      expect(filenameText).toBeInTheDocument();
+      expect(filenameText.closest('a')).toHaveAttribute('href', fileUploadUrl);
+    });
+
+    it('should allow uploading files with the file upload button', async () => {
+      const filename = 'some.txt';
+      const { findByTestId, findByText } = renderComponent({
+        doFileUploadRequest: mockUploadApi(),
+      });
+      const file = new File(['content'], filename, { type: 'text/plain' });
+      const input = (await findByTestId('fileinput')).querySelector('input');
+
+      fireEvent.change(input, {
+        target: {
+          files: [file],
+        },
+      });
+
+      const filenameText = await findByText(filename);
+
+      expect(filenameText).toBeInTheDocument();
+      expect(filenameText.closest('a')).toHaveAttribute('href', fileUploadUrl);
+    });
+
+    it('Should call error handler if an image failed to upload', async () => {
+      const cause = new Error('failed to upload');
+      const doImageUploadRequest = mockFaultyUploadApi(cause);
+      const errorHandler = jest.fn();
+      const { findByPlaceholderText } = renderComponent({
+        doImageUploadRequest,
+        errorHandler,
+      });
+
+      const formElement = await findByPlaceholderText(inputPlaceholder);
+      const file = new File(['(⌐□_□)'], 'chucknorris.png', {
+        type: 'image/png',
+      });
+      dropFile(file, formElement);
+
+      await waitFor(() => {
+        expect(errorHandler).toHaveBeenCalledWith(
+          cause,
+          'upload-image',
+          expect.any(Object),
+        );
+        expect(doImageUploadRequest).toHaveBeenCalledWith(
+          file,
+          expect.any(Object),
+        );
+      });
+    });
+
+    it('Should call error handler if a file failed to upload', async () => {
+      const cause = new Error('failed to upload');
+      const doFileUploadRequest = mockFaultyUploadApi(cause);
+      const errorHandler = jest.fn();
+
+      const { findByPlaceholderText } = renderComponent({
+        doFileUploadRequest,
+        errorHandler,
+      });
+
+      const formElement = await findByPlaceholderText(inputPlaceholder);
+      const file = new File(['content'], 'chucknorris.txt', {
+        type: 'text/plain',
+      });
+      dropFile(file, formElement);
+
+      await waitFor(() => {
+        expect(errorHandler).toHaveBeenCalledWith(
+          cause,
+          'upload-file',
+          expect.any(Object),
+        );
+        expect(doFileUploadRequest).toHaveBeenCalledWith(
+          file,
+          expect.any(Object),
+        );
+      });
+    });
+
+    // TODO: Check if pasting plaintext is not prevented -> tricky because recreating exact event is hard
```

Comment:
Minor: we could use `it.todo('Check if...');` here instead. The nice thing is that it shows on the test report:
![image](https://user-images.githubusercontent.com/1326024/82647090-f3fe5880-9c15-11ea-8071-aa421f34d1ae.png)

Reply:
Ah I had no idea this existed, will add :)

Reply:
Also, agreed about the errors. Those actually don't show up locally for me, so I didn't realise they were in the Travis console output.

I'll look into the best way to get rid of them. One way would be to mock out the `ImagePreview` component (like we discussed). That introduces other difficulties though, because as far as I could tell I could only mock the entire module, which means I would have to reimplement certain things like the dropzone in order to get some of these tests to work. 

Maybe I can just mock out the child-child dependency that actually does the image loading though. I'll investigate a bit more :)